tests/components/ituran/test_config_flow.py

```diff
@@ -0,0 +1,311 @@
+"""Test the Ituran config flow."""
+
+from unittest.mock import AsyncMock
+
+from pyituran.exceptions import IturanApiError, IturanAuthError
+import pytest
+
+from homeassistant.components.ituran.const import (
+    CONF_ID_OR_PASSPORT,
+    CONF_MOBILE_ID,
+    CONF_OTP,
+    CONF_PHONE_NUMBER,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_USER, ConfigFlowResult
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import setup_integration
+from .const import MOCK_CONFIG_DATA
+
+from tests.common import MockConfigEntry
+
+
+async def __do_successful_user_step(
+    hass: HomeAssistant, result: ConfigFlowResult, mock_ituran: AsyncMock
+):
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ID_OR_PASSPORT: MOCK_CONFIG_DATA[CONF_ID_OR_PASSPORT],
+            CONF_PHONE_NUMBER: MOCK_CONFIG_DATA[CONF_PHONE_NUMBER],
+        },
+    )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["step_id"] == "otp"
+    assert result2["errors"] == {}
+
+    return result2
+
+
+async def __do_successful_otp_step(
+    hass: HomeAssistant,
+    result: ConfigFlowResult,
+    mock_ituran: AsyncMock,
+    reauth: bool = False,
+):
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_OTP: "123456",
+        },
+    )
+
+    if reauth:
+        assert result2["type"] is FlowResultType.ABORT
+    else:
```

Comment:
we don't like branching in tests, rather split then. Also, `ABORT` is not a successful step 🙂